### PR TITLE
Use Shriram's env for better upgrade fix.

### PR DESCRIPTION
**Pull request by costinm**
The reminder of the upgrade fix - RDS routes, plus conditionals on the new node environment that 
detects 1.0.

With this a 0.8 sidecar will accept all configs from 1.0 pilot - so prod can be upgraded/rolled without traffic lost. 

For now just using the presence of the env - in 1.1 we may refine this and use per-plugin or capabilities.

As a reminder, upgrade is critical, there are users with 0.8 in production and in google (and everywhere else) ability to gradually roll a new version is required.

### CODE REVIEW

**Comment by rshriram** [comment]
@kyessenov or @mandar-jog ?

File: pilot/pkg/networking/plugin/mixer/mixer.go [hunk 1]
```diff
@@ -228,12 +234,20 @@ func buildInboundHTTPFilter(mesh *meshconfig.MeshConfig, node *model.Proxy, attr
...
 
+func has10Config(node *model.Proxy) bool {
+	_, found := node.Metadata["ISTIO_PROXY_VERSION"]
+	return found
+}
+
 func modifyOutboundRouteConfig(in *plugin.InputParams, httpRoute route.Route) route.Route {
+	if !has10Config(in.Node) {
+		return httpRoute
+	}
```
Comment (rshriram) [inline pilot/pkg/networking/plugin/mixer/mixer.go]: why this short circuiting? we had per-filter config stuff in 0.8

Comment (rshriram) [reply pilot/pkg/networking/plugin/mixer/mixer.go]: @qiwzhang / @kyessenov ^^^ .. I distinctly remember that the mixer was the first to get the per-route filter config.

**Comment by kyessenov** [comment]
Yes, mixer filter supported per route config since last year.

**Comment by costinm** [comment]
```
type.googleapis.com/envoy.api.v2.RouteConfiguration rejected: Unable to parse JSON as proto (INVALID_ARGUMENT:: Cannot find field.): {"forward_attributes":{"attributes":{"destination.service":{"string_value":"istio-ilbgateway.istio-system.svc.cluster.local"},"destination.service.host":{"string_value":"istio-ilbgateway.istio-system.svc.cluster.local"}}},"mixer_attributes":{"attributes":{"destination.service":{"string_value":"istio-ilbgateway.istio-system.svc.cluster.local"},"destination.service.host":{"string_value":"istio-ilbgateway.istio-system.svc.cluster.local"}}},"disable_check_calls":true}

```

**Comment by costinm** [comment]
After enabling the config we're discussing I got the error above. 
How about someone who is familiar with 080 config generation makes a separate PR to generate whatever attributes are needed - maybe per route works but only some of the attributes are supported.

**Comment by kyessenov** [comment]
Forward attributes in per-route service config is the only new field, not
known to 0.8 envoy.

**Comment by costinm** [comment]
Ok, will add it back in separate PR, so this part can get into today build. I doubt lots of people will test the upgrade in next build or notice the missing metrics.

**Pull request merged**